nemo-python/src/lib.rs

```diff
@@ -63,50 +66,116 @@ struct NemoOutputManager(nemo::io::OutputFileManager);
 #[pymethods]
 impl NemoOutputManager {
     #[new]
-    #[pyo3(signature =(path,overwrite=false, gzip=false))]
+    #[pyo3(signature=(path, overwrite=false, gzip=false))]
     fn py_new(path: String, overwrite: bool, gzip: bool) -> PyResult<Self> {
         let output_manager = OutputFileManager::try_new(path.into(), overwrite, gzip).py_res()?;
 
         Ok(NemoOutputManager(output_manager))
     }
 }
 
+#[pyclass]
+#[derive(Debug, PartialEq, Eq)]
+struct NemoLiteral {
+    value: String,
+    language: Option<String>,
+    datatype: String,
+}
+
+#[pymethods]
+impl NemoLiteral {
+    #[new]
+    #[pyo3(signature=(value, lang=None))]
+    fn new(value: PyObject, lang: Option<String>) -> PyResult<NemoLiteral> {
+        Python::with_gil(|py| {
+            let inner: String = value.extract(py).map_err(|_| {
+                NemoError::new_err(format!("Only string arguments are currently supported"))
+            })?;
+
+            Ok(NemoLiteral {
+                value: inner,
+                language: lang,
+                datatype: XSD_STRING.to_string(),
+            })
```

Comment:
This would allow the creation of a `NemoLiteral` with both a datatype and a language tag (which is not supported in either RDF nor in Nemo; in RDF, language-tagged strings always have `rdf:langString` as their implicit datatype). This is probably fine, since the bulk of those will be created from Nemo, but needs a bit if care if, at a later point, we convert a `NemoLiteral` into a primitive logical value.

Reply:
I don't think this is an issue, since this is only our internal representation of the python object that represents an rdf term. It can only be created by users via the `new()` method, which ensures the type of language strings is correctly set.